plugins/outputs/circonus/circonus.go

```diff
@@ -0,0 +1,425 @@
+  # api_insecure_skip_verify = false
+`
+
+var description = "Configuration for Circonus output plugin."
+
+// Metrics values are maps of metric data keyed by metric name.
+type Metrics map[string]Metric
+
+// Metric values describe a single metric data value.
+type Metric struct {
+	Name  string      `json:"-"`
+	Type  string      `json:"_type"`
+	Value interface{} `json:"_value"`
+}
+
+// Conenct creates the client connection to the Circonus broker.
+func (c *Circonus) Connect() error {
+	tlsCfg, err := c.ClientConfig.TLSConfig()
+	if err != nil {
+		return err
+	}
+
+	if tlsCfg == nil {
+		tlsCfg = &tls.Config{}
+	}
+
+	if tlsCfg.RootCAs == nil && !tlsCfg.InsecureSkipVerify {
+		data, err := c.api.Get("/pki/ca.crt")
+		if err != nil {
+			return fmt.Errorf("unable to fetch ca certificate: %w", err)
+		}
+
+		type cacert struct {
+			Contents string `json:"contents"`
+		}
+
+		var cadata cacert
+
+		if err := json.Unmarshal(data, &cadata); err != nil {
+			return fmt.Errorf("error unmarshalling certificate data: %w", err)
+		}
+
+		cp := x509.NewCertPool()
+		if !cp.AppendCertsFromPEM([]byte(cadata.Contents)) {
+			return fmt.Errorf("unable to add Circonus broker CA certificate to certtificate pool")
+		}
+
+		tlsCfg.RootCAs = cp
+	}
+
+	c.client = &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsCfg,
+			Proxy:           http.ProxyFromEnvironment,
+		},
+		Timeout: c.Timeout.Duration,
+	}
+
+	c.apicli = &http.Client{
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+		},
+		Timeout: c.Timeout.Duration,
+	}
+
+	if c.APITLSCA != "" {
+		cert, err := ioutil.ReadFile(c.APITLSCA)
+		if err != nil {
+			return fmt.Errorf("unable to configure Circonus API client: %w", err)
+		}
+
+		cp := x509.NewCertPool()
+		if !cp.AppendCertsFromPEM(cert) {
+			return fmt.Errorf("unable to add Circonus API CA certificate to certtificate pool")
+		}
+
+		c.apicli.Transport = &http.Transport{
+			TLSClientConfig: &tls.Config{RootCAs: cp},
+			Proxy:           http.ProxyFromEnvironment,
+		}
+	}
+
+	return nil
+}
+
+// Write is used to write metric data to Circonus checks.
+func (c *Circonus) Write(metrics []telegraf.Metric) error {
+	mets := map[string]Metrics{}
+
+	for _, m := range metrics {
+		fieldMetrics := c.buildMetrics(m)
+
+		for _, met := range fieldMetrics {
+			for pat, host := range c.Checks {
+				re, err := regexp.Compile(pat)
+				if err != nil {
+					return fmt.Errorf("unable to compile metric regexp: %s: %s",
+						pat, err.Error())
+				}
+
+				if re.MatchString(met.Name) {
+					if mets[host] == nil {
+						mets[host] = Metrics{}
+					}
+
+					mets[host][met.Name] = met
+
+					break
+				}
+			}
+		}
+	}
+
+	autoHost := ""
+	cli := c.client
+
+	for host, ms := range mets {
+		b, err := json.Marshal(ms)
+		if err != nil {
+			return fmt.Errorf("unable to marshal metric data, %s\n",
+				err.Error())
+		}
+
+		if strings.ToLower(host) == "auto" {
+			if autoHost == "" {
```

Comment:
```suggestion
		if strings.ToLower(host) == "auto" && autoHost == "" {
```
saves one indentation level.